src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -107,6 +118,8 @@ public void start(final Map<String, String> parsedConfig) {
     // config as a side effect
     conn = SnowflakeConnectionServiceFactory.builder().setProperties(config).build();
 
+    ingestSdkProvider.createStreamingClient(config, connectorName);
```

Comment:
Alternatively I could have done lazy creation at the task level (first task to start creates the client), but I chose to create here due to possible race conditions. Open to suggestions though

Reply:
This creates a streaming client in a snowpipe case too. I would avoid doing that.

Reply:
Added the check to only create the client for streaming, thanks!